lms/routes.py

```diff
@@ -98,9 +98,16 @@ def includeme(config):  # pylint:disable=too-many-statements
     config.add_route("lti_api.result.read", "/api/lti/result", request_method="GET")
     config.add_route("lti_api.result.record", "/api/lti/result", request_method="POST")
 
-    config.add_route("jstor_api.articles.metadata", "/api/jstor/articles/{article_id}")
+    # JSTOR article IDs need a custom pattern because they may contain a slash,
+    # after URL-decoding of the path.
+    jstor_article_id_pat = r"(10\.[0-9]+/)?[^/]+"
     config.add_route(
-        "jstor_api.articles.thumbnail", "/api/jstor/articles/{article_id}/thumbnail"
+        "jstor_api.articles.metadata",
```

Comment:
We might also get something similar using "remainder match" from https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/urldispatch.html#route-pattern-syntax

But the regex is more explicit :+1:

Reply:
Ah, but a remainder match must come at the end of the path according to those docs. Here it is the "root" endpoint for a JSTOR article and then there are endpoints under that for different kinds of information. One way around that would be to re-organize the path so that the article ID does come at the end, which is how JSTOR's own API is organized. That would be simpler,  but then also different from our typical endpoint structure for other resources.